### ISSUE TITLE
System.Exception: Could not get plugin with specified name () and requested type. - error when downloading from spotify aor youtube music

### ISSUE THREAD

**Issue by bobysmith3** [IcySnex/Melora#3]
![image](https://github.com/user-attachments/assets/3930de7e-f1b5-47ad-bd12-0fb472d1d538)
I have searchged for the same album and clicked download, I went to the downloads and click download all, now I get an error for each track as seen above.

Please advise if you need any further details, also happy to do some testing if needed.

**Comment by IcySnex** [comment]
Hello,
thank you for trying Melora.

This error message seems pretty weird since it will only show this when the plugin isn't loaded.
Are you sure you installed the plugins correctly using [this guide](https://icysnex.github.io/Melora/guide/plugins.html#installing-a-plugin)? Please check if the plugin is shown on the "Plugin Bundles" page.

You can also try restarting Melora and check if the plugins are still loaded correctly. Please make sure to run Melora from a path where you have read/write access like your Desktop folder and do not run it from inside an archive.

Also be aware, that the Spotify plugin is currently not working due to API changes (see #2). It is currently being worked on but YouTube, YouTube Music & SoundCloud should still all work flawlessly.

If the issue persists, please remove Melora, download the latest release, load your plugins and try again. If it still fails, please send me the log files which will be created so I can further investigate. Thank you!

**Comment by IcySnex** [comment]
Hello, since this issue has been inactive for more than 2 weeks, I assume it is resolved. Thus, I will be closing it now. If the problem persists, you can always reopen the issue and I will be happy to help you!

**Comment by mariowitdoek** [comment]
I have the same issue. I reinstalled in a folder c:\melora with full access, FFMPEG.exe in the same folder and the Music in the default folder. I registered YT plugin and when starting, the notification appears. The framework .NET 8 was installed (according powershell check) and I also installed the .NET SDK full with herstart.

2025-02-10 13:00:02.344 +01:00 [INF] [PluginManager-LoadAsync] Starting to load plugin bundle: [C:\Melora\Plugins\Melora.PlatformSupport.YouTube.mlr]...
2025-02-10 13:00:03.551 +01:00 [INF] [YouTubeMusicWrapper-AuthenticateClient] Client has been authenticated.
2025-02-10 13:00:03.552 +01:00 [INF] [YouTubeMusicWrapper-.ctor] YouTubeMusicWrapper has been initialized
2025-02-10 13:00:03.583 +01:00 [INF] [PluginManager-LoadAsync] Loaded plugin: [YouTube Music]
2025-02-10 13:00:03.663 +01:00 [INF] [YouTubeWrapper-AuthenticateClient] Client has been authenticated.
2025-02-10 13:00:03.663 +01:00 [INF] [YouTubeWrapper-.ctor] YouTubeWrapper has been initialized
2025-02-10 13:00:03.665 +01:00 [INF] [PluginManager-LoadAsync] Loaded plugin: [YouTube]
2025-02-10 13:00:03.675 +01:00 [INF] [MainView-ShowNotification] Notification was requested
2025-02-10 13:00:03.682 +01:00 [INF] [PluginBundlesViewModel-ImportAsync] Plugin bundle was imported from file
2025-02-10 13:00:10.177 +01:00 [INF] [PlatformViewModel-.ctor] PlatformViewModel has been initialized
2025-02-10 13:00:10.205 +01:00 [INF] [Navigation-SetPage] Page was set
2025-02-10 13:00:20.437 +01:00 [INF] [PlatformViewModel-SearchAsync] Staring search on YouTube Music...
...
2025-02-10 13:00:27.773 +01:00 [INF] [Navigation-SetCurrentItem] Current navigation item was set
2025-02-10 13:00:27.773 +01:00 [INF] [PlatformViewModel-PrepareDownloadAsync] Prepared downloads on YouTube Music: tony joe white
2025-02-10 13:00:32.542 +01:00 [INF] [DownloadsViewModel-DownloadAsync] Starting download of track
2025-02-10 13:00:32.547 +01:00 [INF] [MainView-ShowNotification] Notification was requested
2025-02-10 13:00:32.547 +01:00 [ERR] [DownloadsViewModel-DownloadAsync] Failed to download track Ain't Going Down This Time: Could not get plugin with specified name () and requested type.
System.Exception: Could not get plugin with specified name () and requested type.
   at Melora.Services.PluginManager.GetLoaded[T](String name)
   at Melora.ViewModels.DownloadsViewModel.DownloadAsync(DownloadContainer download)
2025-02-10 13:00:38.092 +01:00 [INF] [JsonConverter-ToString] Serializing object to string
2025-02-10 13:00:38.129 +01:00 [INF] [MainView-Closed] Closed main window.

**Comment by IcySnex** [comment]
Hello, are you sure you have selected a default [Metadata plugin](https://icysnex.github.io/Melora/guide/metadata.html)? This plugin handles writing metadata like Title, Artist, Image etc. to the audio file.

To check, go to the Melora Settings page, and make sure that "Selected" is not empty under the "Metadata" section.

**Comment by mariowitdoek** [comment]
Indeed, that was the error! I was no so interested in the data inside the files so I did not select them. I did not make the reflex to the Metadata plugin, notwithstanding I read the install instructions very carefully nor the loggings refered to metadata. But knowing this, the error message is indeed logic ...
Vielen Danke & KUTGW!

PS: I have a Spotify subscription so I wouldn't mind a "spotify premium" plugin. It is just because I have a huge collection of CD's and I want to play them on my current audio infrastructure...
